crates/amaru/src/bin/amaru/cmd/bootstrap.rs

```diff
@@ -116,34 +107,41 @@ pub async fn run(args: Args) -> Result<(), Box<dyn Error>> {
 
     import_nonces_for_network(network, &network_dir, &chain_dir).await?;
 
-    import_headers_for_network(network, &args.peer_address, &network_dir, &chain_dir).await?;
+    import_headers_for_network(network, &network_dir, &chain_dir).await?;
 
     Ok(())
 }
 
+#[allow(clippy::unwrap_used)]
 async fn import_headers_for_network(
     network: NetworkName,
-    peer_address: &str,
     config_dir: &Path,
     chain_dir: &PathBuf,
 ) -> Result<(), Box<dyn Error>> {
-    let headers_file: PathBuf = config_dir.join("headers.json");
-    let content = tokio::fs::read_to_string(headers_file).await?;
-    let points: Vec<String> = serde_json::from_str(&content)?;
-    let mut initial_headers = Vec::new();
-    for point in points {
-        match Point::try_from(point.as_str()) {
-            Ok(point) => initial_headers.push(point),
-            Err(e) => tracing::warn!("Ignoring malformed header point '{}': {}", point, e),
+    let era_history = network.into();
+    let mut db = RocksDBStore::new(chain_dir, era_history)?;
+
+    for entry in std::fs::read_dir(config_dir.join("headers"))? {
+        let entry = entry?;
+        let path = entry.path();
+        if path.is_file()
+            && let Some(filename) = path.file_name().and_then(|f| f.to_str())
+            && filename.starts_with("chain.")
+            && filename.ends_with(".cbor")
+        {
+            let mut file = File::open(&path).await
+                    .inspect_err(|reason| tracing::error!(file = %path.display(), reason = %reason, "Failed to open header file"))
+                    .map_err(|_| WorkerError::Panic)?;
+            let mut cbor_data = Vec::new();
+            file.read_to_end(&mut cbor_data).await
+                    .inspect_err(|reason| tracing::error!(file = %path.display(), reason = %reason, "Failed to read header file"))
+                    .map_err(|_| WorkerError::Panic)?;
+            let header_from_file: Header = from_cbor(&cbor_data).unwrap();
+            let hash = header_from_file.hash();
+            db.store_header(&hash, &header_from_file)
+                .map_err(|_| WorkerError::Panic)?;
```

Comment:
Why WorkerError?

Reply:
Because we're using gasket. If that's ok for you, can you resolve this conversation and we keep your comment above about not using gasket?